src/software/simulated_tests/simulated_test_fixture.cpp

```diff
@@ -167,10 +167,23 @@ void SimulatedTestFixture::runTest(
     const Duration ai_time_step = Duration::fromSeconds(simulation_time_step.toSeconds() *
                                                         CAMERA_FRAMES_PER_AI_TICK);
 
+    auto start_tick_time = std::chrono::system_clock::now();
+
     // Tick one frame to aid with visualization
     bool validation_functions_done = tickTest(simulation_time_step, ai_time_step, world);
+
+    // Logging duration of each tick
+    unsigned int tick_count    = 1;
+    double duration_ms         = ::TestUtil::millisecondsSince(start_tick_time);
+    double total_tick_duration = duration_ms;
+    double max_tick_duration   = duration_ms;
+    double min_tick_duration   = duration_ms;
+
     while (simulator->getTimestamp() < timeout_time && !validation_functions_done)
     {
+        // Record starting time
+        start_tick_time = std::chrono::system_clock::now();
```

Comment:
this should be after we sleep, i.e. right before tickTest

Reply:
Updated